### DISCUSSION TITLE
Suddenly, I got the error `fatal: unable to access 'https://github.com/xxx/yyy.git/': The requested URL returned error: 403`

### DISCUSSION THREAD

**Discussion by monagai** [community/community#82804] [Repositories] [answered]
### Select Topic Area

General

### Body

I got suddenly 403 error with  `git clone https://github.com/xxx/yyy.git`.
Until several days ago, it has not caused no error.

The situation is as foolows:
+ Git clone is OK for public repositories.
+ Git clone is OK for private repositories in 2 organizations.
  I am a one of the owner of one of these organaizeions.
+ Git clone retuns 403 error for private repositories in 1 organization. 
  I am one of owners of this organization.
+ I'm using a classic token. It will expire Jul. 2024.
  I tried to use new tokens. However the situation does not differ.
+ I'm using only 1 account.
+ I tried in two environments. The situaion does not differ.
  - Cent OS 7
  - Mac OS 12.7.2
+ I ask another owner of this organization to remove me and send invitation to me again.
  However, the situation does not differ.
+ I don't think someone changed the setting of this organization.

How shoud I do?
**It is really fatal to me.**

**Comment by ananuness** [comment]
The [Github CLI](https://docs.github.com/en/get-started/getting-started-with-git/caching-your-github-credentials-in-git) can help you to authenticate and always remember your credentials as you are using HTTPS. Have you tried it?

**Reply by monagai** [reply]
> The [Github CLI](https://docs.github.com/en/get-started/getting-started-with-git/caching-your-github-credentials-in-git) can help you

I really thank you for your kind and quick advice. With your advice, I tried it in my Mac for the first time.
```
gh auth login
? What account do you want to log into? GitHub.com
? What is your preferred protocol for Git operations on this host? HTTPS
? Authenticate Git with your GitHub credentials? Yes
? How would you like to authenticate GitHub CLI? Paste an authentication token
Tip: you can generate a Personal Access Token here https://github.com/settings/tokens
The minimum required scopes are 'repo', 'read:org', 'workflow'.
? Paste your authentication token: ****************************************
error validating token: HTTP 401: Bad credentials (https://api.github.com/)
Try authenticating with:  gh auth login
```
Does it mean my token is bad?
Why can I run `git clone` successfully with private repositories in other 2 organizations???
Yesterday when I tried to use regenerated new token, the situation didn't differ.  Wny? What is wrong?

**Reply by ananuness** [reply]
Ok, that's strange if you're an owner and your [token](https://github.com/settings/tokens) is fine and work well. I'm trying to search more about it, but I just found a fact that may help you, check [those steps](https://docs.github.com/en/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors#https-cloning-errors) to guarantee you're not having trouble with the org or your token.

**Reply by monagai** [reply]
@ananuness 
Thank you for your kind advice. I will try to follow your advice though I have not understood fully your comment.

By the way, I tried `gh auth login` with new classic token. This new token works.
```
$ gh auth login
? What account do you want to log into? GitHub.com
? What is your preferred protocol for Git operations on this host? HTTPS
? Authenticate Git with your GitHub credentials? Yes
? How would you like to authenticate GitHub CLI? Paste an authentication token
Tip: you can generate a Personal Access Token here https://github.com/settings/tokens
The minimum required scopes are 'repo', 'read:org', 'workflow'.
? Paste your authentication token: ****************************************
- gh config set -h github.com git_protocol https
✓ Configured git protocol
✓ Logged in as monagai
```
So, I deleted the old token.
However,  still I cannot git clone for one of the organizations.
I'm really confused.

**Reply by ananuness** [reply]
I'm sorry, I misunderstood the token stuff, I was reading about and suggested something about oauth token, not personal access token 😅 but about the organization, maybe the organization has enabled [SAML single sign-on](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-saml-single-sign-on-for-your-organization/about-identity-and-access-management-with-saml-single-sign-on) (if you're using github enterprise cloud), then access tokens need to be authorized! If not, I didn't find anything that could block the access to an org yet. Either way, check out the [org settings](https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/setting-a-personal-access-token-policy-for-your-organization)

**Reply by monagai** [reply, marked as answer]
Dear @ananuness 
Thank you for you kind and quick advice and I'm really sorry for my lack of information (sorry, I don't understand github security system much).
With you advice "check out the [org settings](https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/setting-a-personal-access-token-policy-for-your-organization)", I solved the problem!!
The organization setting about personal access toeken was set "Restrict access via personal access tokens (classic)"!
(Another owner has changed the setting??) 
I changed it to "Allow access via personal access tokens (classic)" and the problem was solved.
I could solve the problem with your help. I deeply thank you again. You have spent much of labor for me!

**Reply by ananuness** [reply]
Yaay! I'm happy we could solve this! And no problem, I misunderstood some things hehe thank you for your patience! Remember to mark your answer, it will help others with the same problem!

**Reply by juanluisoc** [reply]
No sería mas fácil de sirmelo directamente libertad y no andar con rodeos ya sabes que ami me gusta ablar derecho y todo eso causionastes todo el preocupaciones de mi gente todos angustiados por que tu aceptastes segurles la corriente a eyos te importo mas eyos que to y mi familia 
Yo no me la se así yo me la se de otra manera pero aya tu ya sabes que todo se sabe tarde o temprano ya me bas andar la cara on Ban asegurar con sus secretos todavía o asta cuando ba terminar esto asta que me maten eso es lo que estan es perando barios así labeo las cosas con eso de sebes que no les cuento a nadien de mis cosas ni de mis inversiones pero yo ya me perebine con lo que me fuera a pasar ya saben con quien irse directo saben la istoria de todo lo que me an echo o Ban a esperar asta que me que de pobre otra bes ya me an robado mucho inversiones y carteras que no cres que me edad cuenta de lo que yo yebaba mis cuentas pero se a sen de que nadie sabe nada y estan amenazadas a que no me ayuden en la contabilidad pero si alguna persona les paso todo mis datos y mis cuentas a eyos pero por que razón por darme o matarme si yo las tengo des de ase mucho tiempo eso yo lo empecé y me las conastes aso tu me dise pero enfin solo dame la cara y ablemos ya yano mas por mensages ni por recados por que ya no me la creo oy en dia ay muchos estafadores por que ya me la an echo muchas beses y a lultimo serien de mi para que yo le asada las ---autorice--- pero ya no mas grasia te be mañana por la tarde o oy mismo tu decides
